src/components/rangeslider/defaults.js

```diff
@@ -12,14 +12,13 @@ var Lib = require('../../lib');
 var attributes = require('./attributes');
 
 
-module.exports = function supplyLayoutDefaults(layoutIn, layoutOut) {
+module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, axName, counterAxes) {
 
-    if(!layoutIn.xaxis || !layoutOut.xaxis) return;
+    if(!layoutIn[axName].rangeslider) return;
```

Comment:
I see your point. But at the moment, container attribute objects are either objects or undefined and nothing else.